### PR TITLE
Do safety check before detecting hex value and handle YAML numbers better

### PR DESCRIPTION
**Pull request by KTibow**
## Proposed change

This does a safety check to make sure the value exists before checking if it's hex, so you don't get errors like `Uncaught (in promise) TypeError: t.startsWith is not a function`. It also handles numbers better if you forgot to quote them.

## Type of change

- [ ] Dependency upgrade
- [x] Bugfix (non-breaking change which fixes an issue)
- [ ] New feature (thank you!)
- [ ] Breaking change (fix/feature causing existing functionality to break)
- [ ] Code quality improvements to existing code or addition of tests

## Example configuration

```yaml
frontend:
  themes:
    happy:
      ha-card-header-font-weight: 600
```

## Additional information

- This PR fixes or closes issue: fixes #
- This PR is related to issue:
- Link to documentation pull request:

## Checklist

- [ ] The code change is tested and works locally.
- [x] There is no commented out code in this PR.
- [ ] Tests have been added to verify that the new code works.

If user exposed functionality or configuration variables are added/changed:

- [ ] Documentation added/updated for [www.home-assistant.io][docs-repository]

[docs-repository]: https://github.com/home-assistant/home-assistant.io

### CODE REVIEW

File: src/common/dom/apply_themes_on_element.ts [hunk 1]
```diff
@@ -110,7 +110,7 @@ const processTheme = (
     keys[prefixedKey] = "";
 
     // Try to create a rgb value for this key if it is not a var
-    if (!value.startsWith("#")) {
+    if (!value || !value.startsWith("#")) {
```
Comment (bramkragten) [inline src/common/dom/apply_themes_on_element.ts:113]: In what situation is this undefined?

You should use `value?.startsWith(...`

**Review by bramkragten — CHANGES_REQUESTED** [review]

File: src/common/dom/apply_themes_on_element.ts (hunk 1 above)
Comment (KTibow) [reply src/common/dom/apply_themes_on_element.ts:113]: Really, this PR is pointless. I was having trouble getting changing the font weight to work, but then I figured out that if I wrapped it in quotes it worked fine. I'm going to go ahead and add that in and edit.

Comment (KTibow) [reply src/common/dom/apply_themes_on_element.ts]: Anyway, added that and that.

File: src/common/dom/apply_themes_on_element.ts [hunk 2]
```diff
@@ -105,12 +105,12 @@ const processTheme = (
...
   for (const key of Object.keys(combinedTheme)) {
     const prefixedKey = `--${key}`;
-    const value = combinedTheme[key]!;
+    const value = String(combinedTheme[key]!);
     styles[prefixedKey] = value;
     keys[prefixedKey] = "";
 
     // Try to create a rgb value for this key if it is not a var
-    if (!value.startsWith("#")) {
+    if (value?.startsWith("#")) {
```
Comment (bramkragten) [inline src/common/dom/apply_themes_on_element.ts:113]: If we make it a string this is not needed
```suggestion
    if (value.startsWith("#")) {
```

**Pull request merged**